### ISSUE TITLE
FOQUS Flowsheet: Deactivating an edge does not give intended results

### ISSUE THREAD

**Issue by fsoepyan** [CCSI-Toolset/FOQUS#335]
Results of a FOQUS run are different for the case of deactivating an edge, and deleting that same edge. However, in theory, these results should be the same.

**Comment by anujad95** [comment]
Is this still an issue? Also, is there a particular example that was implemented, and did not work in this case?

**Comment by anujad95** [comment]
I tried a simple example where i connected 2 nodes with an edge, through the output & input variables of each one respectively. It seems that deactivating and deleting the edge has the same effect. Once the flowsheet is run with an active edge, the input values of the 2nd node are automatically populated with the corresponding output values of the 1st node. So, if the edge is deactivated after this, you may want to change some input values of the 1st node to see the effect. The new output values of 1st node would not affect the 2nd node input values after deactivation.

**Comment by eslickj** [comment]
I'm not sure about an example, but I think the issue is with finding tear streams.  I think all or part of that code ignores the fact that an edge is deactivated, and treats it as if it is active.  You could try on of the simple tear stream examples, and see if it works right.

The fix isn't hard, but it may get complicated to since the tear finding code is a bit complicated.  We may need to rework the flowsheet for provide some function that returns active edges and update everything.  The first step may be to write some tests, just to make sure we don't break anything else when fixing this.

**Comment by fsoepyan** [comment]
I tried to deactivate an edge for this simple example:

![image](https://user-images.githubusercontent.com/30811140/60536596-96e37680-9cd4-11e9-9ea0-e4a21313641a.png)

When I deactivated the edge connecting "example" to "second", the Flowsheet runs as intended.

I will test the edge deactivation using an example involving a tear stream.

**Comment by anujad95** [comment]
I can check a simple tear stream example and get back to you on this. Thanks.

**Comment by fsoepyan** [comment]
- I tested the edge deactivation using this simple example involving a tear stream:

![image](https://user-images.githubusercontent.com/30811140/60539832-a1edd500-9cdb-11e9-89ae-b48684eb2d40.png)

- Whichever edge I deactivated, the simulation works.
- However, when I tried to re-activate the tear stream automatically by clicking the tear stream button on the left-side menu, initially, it automatically deactivated one of the edges.
- To fix this issue, I manually checked the boxes to activate both edges, and to assign one of the edges as a tear stream.
- In other words, if the user wants to deactivate an edge for simulations involving tear streams, the user will need to remember which edges are the tear streams.

**Comment by fsoepyan** [comment]
- I redid the above tear stream example.
- I tried deactivating both edges, and deactivating only 1 edge at a time.
- The simulations work for all 3 cases.
- Then, I manually checked the boxes next to "Activate" for both edges in the Edge Editor.
- When I clicked the tear stream button on the left-side menu, the correct edge is identified as the tear stream.

**Comment by fsoepyan** [comment]
Given my testing and comments above, can we close this issue?

**Comment by anujad95** [comment]
Yes, it seems to be fine. According to your tests, we don't need any source code modifications right?

**Comment by fsoepyan** [comment]
Correct.
